src/kex_sidh_cln16/kex_sidh_cln16.c

```diff
@@ -17,18 +17,15 @@
 #include "SIDH.h"
 #include "kex_sidh_cln16.h"
 
-static char *P751 = "p751";
-static char *CompressedP751 = "compressedp751";
+static const char *P751 = "p751";
+static const char *CompressedP751 = "compressedp751";
 
 static int isCompressed(const char *named_parameters) {
-	int compressed = 0; // defaults to non-compressed
-	if (named_parameters == NULL ||
-	    strncmp(P751, named_parameters, strlen(P751)) == 0) {
-		compressed = 0;
-	} else if (strncmp(CompressedP751, named_parameters, strlen(CompressedP751)) == 0) {
-		compressed = 1;
-	}
-	return compressed;
+	if (named_parameters != NULL && strcmp(named_parameters, CompressedP751) == 0) {
```

Comment:
Can you use strncmp like in the original code?

Reply:
I'd be happy to change it to use strncmp, but could you please elaborate a bit on the reason for checking only the beginning of the parameter string? IMO expected behavior for similar checks is to do exact matching. In the original code, if you try to use "p751compressed" (which is an invalid parameter), it would match on "p751" and use that.

Reply:
You are right that the original code had this quirk. I prefer in general to use the strncmp, but since we use strcmp elsewhere in OQS, I'm fine with its use here too.